heka/plugins/s3splitfile/s3splitfile_input.go

```diff
@@ -39,27 +40,29 @@ type S3SplitFileInputConfig struct {
 	// So we can default to using HekaFramingSplitter.
 	Splitter string
 
-	SchemaFile     string `toml:"schema_file"`
-	AWSKey         string `toml:"aws_key"`
-	AWSSecretKey   string `toml:"aws_secret_key"`
-	AWSRegion      string `toml:"aws_region"`
-	S3Bucket       string `toml:"s3_bucket"`
-	S3BucketPrefix string `toml:"s3_bucket_prefix"`
-	S3Retries      uint32 `toml:"s3_retries"`
-	S3WorkerCount  uint32 `toml:"s3_worker_count"`
+	SchemaFile             string `toml:"schema_file"`
+	AWSKey                 string `toml:"aws_key"`
+	AWSSecretKey           string `toml:"aws_secret_key"`
+	AWSRegion              string `toml:"aws_region"`
+	S3Bucket               string `toml:"s3_bucket"`
+	S3BucketPrefix         string `toml:"s3_bucket_prefix"`
+	S3ObjectBasenamePrefix string `toml:"s3_object_basename_prefix"`
```

Comment:
Should we add a basename "pattern" instead of a simple "prefix"?  It should be about the same complexity, but much more powerful.